node/node_explorer.go

```diff
@@ -123,7 +123,7 @@ func (node *Node) explorerMessageHandler(ctx context.Context, msg *msg_pb.Messag
 func (node *Node) AddNewBlockForExplorer(block *types.Block) {
 	utils.Logger().Info().Uint64("blockHeight", block.NumberU64()).Msg("[Explorer] Adding new block for explorer node")
 	if _, err := node.Blockchain().InsertChain([]*types.Block{block}, true); err == nil {
-		if len(block.Header().ShardState()) > 0 {
+		if block.Header().IsLastBlockInEpoch() {
```

Comment:
you already have block.IsLastBlockInEpoch

Reply:
yep forgot to change this